src/main/db_main.cpp

```diff
@@ -21,63 +25,68 @@ void DBMain::Init() {
           param_map_.find(settings::Param::record_buffer_segment_size)->second.value_),
       type::TransientValuePeeker::PeekInteger(
           param_map_.find(settings::Param::record_buffer_segment_reuse)->second.value_));
-  txn_manager_ = new transaction::TransactionManager(buffer_segment_pool_, true, nullptr);
-  gc_thread_ = new storage::GarbageCollectorThread(txn_manager_,
-                                                   std::chrono::milliseconds{type::TransientValuePeeker::PeekInteger(
-                                                       param_map_.find(settings::Param::gc_interval)->second.value_)});
-  transaction::TransactionContext *txn = txn_manager_->BeginTransaction();
   settings_manager_ = new settings::SettingsManager(this);
-  txn_manager_->Commit(txn, transaction::TransactionUtil::EmptyCallback, nullptr);
+  thread_registry_ = new common::DedicatedThreadRegistry;
 
   // Create LogManager
   log_manager_ = new storage::LogManager(
       settings_manager_->GetString(settings::Param::log_file_path),
       settings_manager_->GetInt(settings::Param::num_log_manager_buffers),
       std::chrono::milliseconds{settings_manager_->GetInt(settings::Param::log_serialization_interval)},
       std::chrono::milliseconds{settings_manager_->GetInt(settings::Param::log_persist_interval)},
-      settings_manager_->GetInt(settings::Param::log_persist_threshold), buffer_segment_pool_);
+      settings_manager_->GetInt(settings::Param::log_persist_threshold), buffer_segment_pool_,
+      common::ManagedPointer(thread_registry_));
   log_manager_->Start();
 
+  txn_manager_ = new transaction::TransactionManager(buffer_segment_pool_, true, log_manager_);
```

Comment:
Curious, what was the reasoning to changing the ordering of what order we bring up the components?

Reply:
`TransactionManager` was being brought up before `LogManager` so logging was never enabled.